x-template.js

```diff
@@ -470,8 +470,12 @@ class TemplateEngine {
       const index = values.length;
       const id = String(index);
       const item = arrayState.map.get(id);
-      TemplateEngine.#removeThrough(item.startNode, node);
-      arrayState.map.delete(id);
+      TemplateEngine.#removeBetween(item.startNode, node);
+      item.startNode.remove();
+      for (let iii = arrayState.map.size - 1; iii >= values.length; iii--) {
+        // We iterate backwards since we are deleting keys from the map itself.
+        arrayState.map.delete(String(iii));
```

Comment:
And, we need to make sure to property truncate the _map_. The alternative would be to keep an `array` and splice it to remove values… I’m not actually convinced that would be an overall performance win though since you are then forced to use array-access elsewhere.

Anyways, this was the simplest fix I could implement and we can do another performance pass in the future. I just really wanted to lock this stuff down in our test suite.